lib/commands/context.js

```diff
@@ -184,6 +192,32 @@ helpers.startChromedriverProxy = async function startChromedriverProxy (context)
       if (androidPackage && androidPackage.length > 0) {
         opts.chromeAndroidPackage = androidPackage[1];
       }
+      if (!opts.extractChromeAndroidPackageFromContextName) {
+        if (_.has(this.opts, 'enableWebviewDetailsCollection') && !this.opts.enableWebviewDetailsCollection) {
+          // When enableWebviewDetailsCollection capability is explictly disabled, try to identify
+          // chromeAndroidPackage based on contexts, known chrome variant packages and queryAppState result
+          // since webviewsMapping does not have info object
+          const contexts = webviewsMapping.map((wm) => wm.webviewName);
+          for (const knownPackage of KNOWN_CHROME_PACKAGE_NAMES) {
+            if (_.includes(contexts, `${WEBVIEW_BASE}${knownPackage}`)) {
+              continue;
+            }
+            const appState = await this.queryAppState(knownPackage);
+            if (appState === 3 || appState === 4) {
+              // When APP_STATE_RUNNING_IN_BACKGROUND or APP_STATE_RUNNING_IN_FOREGROUND
+              opts.chromeAndroidPackage = knownPackage;
+              break;
+            }
+          }
+        } else {
+          for (const wm of webviewsMapping) {
+            if (wm.webviewName === context && _.has(wm?.info, 'Android-Package')) {
+              opts.chromeAndroidPackage = wm.info['Android-Package'];
```

Comment:
having some debug logging here might be helpful

Reply:
Addressed as https://github.com/appium/appium-android-driver/pull/699/commits/4a83de0aa9a1781325e462dd26a04b90873ace9a